dbt/adapters/trino/catalogs/_trino_catalog_metastore.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+from dbt.adapters.catalogs import CatalogIntegration, CatalogIntegrationConfig
+from dbt.adapters.contracts.relation import RelationConfig
+
+from dbt.adapters.trino.catalogs._relation import TrinoCatalogRelation
+
+
+class TrinoCatalogIntegration(CatalogIntegration):
+    # metastore is defined at Trino catalog configuration
+    # https://trino.io/docs/current/overview/concepts.html#catalog
+    catalog_type = "trino"
+    allows_writes = True
+    # table_format is available for Galaxy
+    # https://docs.starburst.io/starburst-galaxy/data-engineering/working-with-data-lakes/table-formats/index.html
+    # In Trino or SEP table format is specified by which connector is used
+    table_format = None
+    file_format = None
+
+    def __init__(self, config: CatalogIntegrationConfig) -> None:
+        super().__init__(config)
+        self.storage_uri = config.adapter_properties.get("storage_uri")
+
+    def build_relation(self, model: RelationConfig) -> TrinoCatalogRelation:
+        return TrinoCatalogRelation(
+            catalog_type=self.catalog_type,
+            catalog_name=self.catalog_name,
+            table_format=self.table_format,
+            file_format=self.file_format,
+            external_volume=self.external_volume,
+            storage_uri=self._calculate_storage_uri(model),
+        )
+
+    def _calculate_storage_uri(self, model: RelationConfig) -> Optional[str]:
+        if not model.config:
+            return None
+
+        if model_storage_uri := model.config.get("storage_uri"):
+            return model_storage_uri
+
+        if not self.external_volume:
+            return None
+
+        prefix = model.config.get("base_location_root") or "_dbt"
+        omit_base_location_root = model.config.get("omit_base_location_root")
```

Comment:
Can you add some documentation somewhere about what omit_base_location_root is for an its behavior.  Also specify that if this is set to true, any value set in `base_location_root` will be ignored